source/OutgoingMessages.Infrastructure/Databricks/WholesaleResults/Queries/WholesaleTotalAmountQuery.cs

```diff
@@ -103,17 +106,24 @@ protected override Task<WholesaleTotalAmountMessageDto> CreateWholesaleResultAsy
     }
 
     private static (ActorNumber ActorNumber, ActorRole ActorRole) GetReceiver(
-        DatabricksSqlRow databricksSqlRow)
+        DatabricksSqlRow databricksSqlRow,
+        ImmutableDictionary<string, ActorNumber> gridAreaOwnerDictionary)
     {
         var chargeOwnerNumber = GetChargeOwnerNumber(databricksSqlRow);
+        var gridAreaCode = databricksSqlRow.ToNonEmptyString(WholesaleResultColumnNames.GridAreaCode);
         var energySupplierNumber =
             ActorNumber.Create(databricksSqlRow.ToNonEmptyString(WholesaleResultColumnNames.EnergySupplierId));
+
         if (chargeOwnerNumber is not null)
         {
+            if (chargeOwnerNumber != DataHubDetails.SystemOperatorActorNumber)
+            {
+                var gridAreaOwner = gridAreaOwnerDictionary[gridAreaCode];
+                return (gridAreaOwner, GetChargeOwnerRole(gridAreaOwner));
```

Comment:
```suggestion
                return (gridAreaOwner, ActorRole.GridAccessProvider);
```

Would it be more explicit that in the if case that the actor is not the `SystemOperatorActorNumber` it must be a GridAccessProvider?

I would suggest we remove `GetChargeOwnerRole()` to make the code easier to read.